src/Compilers/CSharp/Portable/Binder/BlockBinder.cs

```diff
@@ -10,27 +10,27 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     internal sealed class BlockBinder : LocalScopeBinder
     {
-        private readonly SyntaxList<StatementSyntax> _statements;
+        private readonly BlockSyntax _block;
 
-        public BlockBinder(Binder enclosing, SyntaxList<StatementSyntax> statements)
-            : this(enclosing, statements, enclosing.Flags)
+        public BlockBinder(Binder enclosing, BlockSyntax block)
+            : this(enclosing, block, enclosing.Flags)
         {
         }
 
-        public BlockBinder(Binder enclosing, SyntaxList<StatementSyntax> statements, BinderFlags additionalFlags)
+        public BlockBinder(Binder enclosing, BlockSyntax block, BinderFlags additionalFlags)
             : base(enclosing, enclosing.Flags | additionalFlags)
         {
-            _statements = statements;
+            _block = block;
```

Comment:
> Is it okay for _block to be null? 

No.

> If not should we assert here?

I didn't think it would be valuable since we don't introduce new places where this binder is created and all those places used to dereference block anyway. Do you think the assert will add value here?